From 6506c92b24744fb77dd9446e5a34e1d03d95d1c0 Mon Sep 17 00:00:00 2001
From: Varsha Maira <vmaira@habana.ai>
Date: Wed, 21 May 2025 10:15:25 +0300
Subject: [PATCH] [SW-229103] g3: reduce spi fifo threshold level to fix tpm failures

TPM transactions are failing intermittently due to SPI RX buffer overflow
at the Gaudi3 SPI controller. Reducing the TPM SPI frequency leads to TX
buffer overflows.

This patch reduces the 'fifo-depth' property in the device tree for the
TPM SPI controller on gaudi3. Since the FIFO threshold levels (RXFTLR and
TXFTLR) are derived from fifo-depth, lowering it indirectly reduces these
threshold levels as well.

Reducing the FIFO thresholds causes 'FIFO full' interrupts to trigger more
frequently. This ensures RX buffer reads happen faster and TX buffer
writes are delayed slightly, helping to avoid overflows.

This is a temporary workaround until the root cause of the buffer
overflows is identified.

Change-Id: I1b97ba8a4f2a533f223fe7c4fa175957b069a923
Signed-off-by: Varsha Maira <vmaira@habana.ai>
---

diff --git a/dts/arc/hl_gaudi3.dtsi b/dts/arc/hl_gaudi3.dtsi
index 174ddd6..72a90a0 100644
--- a/dts/arc/hl_gaudi3.dtsi
+++ b/dts/arc/hl_gaudi3.dtsi
@@ -787,7 +787,7 @@
 			reg = <0xfc483000 0x100>;
 			clocks = <&refclk>;
 			interrupts = <323 1>;
-			fifo-depth = <16>;
+			fifo-depth = <8>;
 			max-xfer-size = <32>;
 			cs-gpios = <&gpio1 2 GPIO_ACTIVE_LOW>;
 			node_label = "SPI_1";
